app/controllers/AdminController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+//require_once 'models/PostModel.php';
+//require_once 'models/CommentModel.php';
+//require_once 'models/UserModel.php';
+namespace app\controllers;
+use app\models\CommentModel;
+use app\models\PostModel;
+use app\models\UserModel;
+use app\utils\View;
+
+class AdminController {
+
+    private PostModel $postModel;
+    private CommentModel $commentModel;
+    private UserModel $userModel;
+
+    public function __construct() {
+        $this->postModel = new PostModel();
+        $this->commentModel = new CommentModel();
+        $this->userModel = new UserModel();
+    }
+
+    public function index(): void
+    {
+        $posts = $this->postModel ->getPosts();
+//        require_once 'views/admin/dashboard.php';
+        View::render('views/admin/dashboard.php', ['posts' => $posts]);
+    }
+
+    public function show($id): void
+    {
+        $post = $this->postModel->getPostByID($id);
+        $comments = $this->commentModel->getApprovedComments($id);
+        if ($post == null) {
+            $post = [];
+        }
+        if ($comments == null) {
+            $comments = [];
+        }
+//        var_dump($id);
+//        var_dump($post);
+//        var_dump($comment);
+//        require_once 'views/admin/admin_show.php';
+        View::render('views/admin/admin_show.php', ['post' => $post, 'comments' => $comments]);
+    }
+
+    public function users(): void
+    {
+        $users = $this->userModel->findAllUsers();
+//        require_once 'views/admin/usermanagement.php';
+        View::render('views/admin/usermanagement.php', ['users' => $users]);
+    }
+
+    public function create(): void
+    {
+        if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+            $username = $_POST['username'];
```

Comment:
@tony-tang-aligent what would happen here if any of these POST variables weren't set?

Reply:
Thanks for the tip, really good hints. I should have added in error handling logic in the else clause, if the request method is not POST or if any of these POST variables weren't set, should report an error message or redirect to 404? I was thinking of sending an error message or using try catch to handle the else clause

Reply:
I think a 400 status code would be appropriate here.